src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/NativeObjCMarshalTests.cpp

```diff
@@ -47,20 +47,18 @@ namespace
             ::abort();
         }
 
-
         assert(mem != nullptr);
         auto cxt = (Contract*)mem;
 
-        cxt->RefCountDown--;
-        cxt->RefCountUp++;
-
         if (cxt->RefCountDown == 0)
         {
             // No more references
             return 0;
         }
         else
         {
+            cxt->RefCountDown--;
+            cxt->RefCountUp++;
```

Comment:
The callback is going to return that there's no references one call later than before - this is fine because we always explicitly trigger the GC more than the initial count, right? And the test relies on the initial count being at least one less than the number of times we explicitly trigger the GC (otherwise it will be unreliable)?
https://github.com/dotnet/runtime/blob/1f3d36a323c9efb5e9a0f1f539332cc745c90c21/src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/Program.cs#L152
https://github.com/dotnet/runtime/blob/1f3d36a323c9efb5e9a0f1f539332cc745c90c21/src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/Program.cs#L208
If so, could we add some sort of comment or make const variable to make it clear those are connected for the test to be reliable?

Reply:
Sure. I will add a comment.